imap_processing/ena_maps/ena_maps.py

```diff
@@ -873,22 +881,30 @@ def project_pset_values_to_map(
                     value_array=raveled_pset_data,
                     projection_grid_shape=self.binning_grid_shape,
                     projection_indices=matched_indices_push,
+                    input_valid_mask=valid_mask,
                 )
+                # TODO: we may need to allow for unweighted/weighted means here by
+                # dividing pointing_projected_values by some binned weights.
+                # For unweighted means, we could use the number of pointing set pixels
+                # that correspond to each map pixel as the weights.
+                self.data_1d[value_key] += pointing_projected_values
             elif index_match_method is IndexMatchMethod.PULL:
                 # We know that there will only be one value per sky map pixel,
                 # so we can use the matched indices directly
                 pointing_projected_values = raveled_pset_data[..., matched_indices_pull]
+                # TODO: we may need to allow for unweighted/weighted means here by
+                # dividing pointing_projected_values by some binned weights.
+                # For unweighted means, we could use the number of pointing set pixels
+                # that correspond to each map pixel as the weights.
+                valid = valid_mask[matched_indices_pull]
+                self.data_1d[value_key].values[..., valid] += pointing_projected_values[
+                    ..., valid
+                ]
```

Comment:
This is a minor nit, but you could avoid some overhead by only pulling valid values:
```suggestion
                valid_map_mask = valid_mask[matched_indices_pull]
                pointing_projected_values = raveled_pset_data[..., matched_indices_pull[valid_map_mask]]
                # TODO: we may need to allow for unweighted/weighted means here by
                # dividing pointing_projected_values by some binned weights.
                # For unweighted means, we could use the number of pointing set pixels
                # that correspond to each map pixel as the weights.
                
                self.data_1d[value_key].values[..., valid_map_mask] += pointing_projected_values
```